gematria/granite/convert_gb_token_model_to_tflite.sh

```diff
@@ -78,19 +86,23 @@ function str_join() {
 }
 
 # The list of inputs of the model. This must contain an entry for each
-# tf.placeholder tensor used in the Python code. The order of the tensors used
-# here must correspond to the input tensor indices defined in
-# gematria/granite/graph_builder_model_inference.cc.
+# tf.placeholder tensor used in the Python code.
 readonly INPUT_TENSORS_LIST=(
-  ModelBase.delta_block_index_tensor
+  $([[ "${gematria_export_as_seq2seq}" -eq 1 || \
+       "${gematria_export_with_deltas}" -eq 1 ]] && \
+    echo 'ModelBase.delta_block_index_tensor')
   GnnModelBase.node_features
   GnnModelBase.edge_features
   GnnModelBase.global_features
   GnnModelBase.receivers
   GnnModelBase.senders
   GnnModelBase.num_edges
   GnnModelBase.num_nodes
-  GraphBuilderModelBase.instruction_node_mask
+  $([[ "${gematria_export_with_annotations}" -eq 1 ]] && \
+    echo 'GraphBuilderModelBase.instruction_node_mask')
+  $([[ "${gematria_export_with_deltas}" -eq 1 || \
+       "${gematria_export_with_annotations}" -eq 1 ]] && \
+    echo 'TokenGraphBuilderModel.instruction_annotations')
 )
```

Comment:
A more readable version would do something like
```bash
declare -a INPUT_TENSORS_LIST=()
if ...; then
  INPUT_TENSORS_LIST+=(...)
fi
INPUT_TENSORS_LIST+=(...)
...
readonly INPUT_TENSORS_LIST
```

This way, the list is read-only in the end, but we can use simpler way to compose it. Also, ideally, we should use just tensor names and do the index matching from there, but that's for another series of patches.

Reply:
I initially chose this (admittedly strange) way of doing this as a concise way of letting optional tensors be at any position since IIRC the ordering of tensors needed to match that of the tflite model (though this PR also includes name-based matching, which should eliminate that requirement).

Reply:
The += syntax would still allow us to add at any spot in the list (and we'd avoid the inline syntax with echos):
```sh
declare -a INPUT_TENSORS_LIST=()
if (( "${gematria_export_as_seq2seq}" )); then
  INPUT_TENSORS_LIST+=(ModelBase.delta_block_index_tensor)
fi
INPUT_TENSORS_LIST+=(
  GnnModelBase.node_features
  GnnModelBase.edge_features
  GnnModelBase.global_features
  GnnModelBase.receivers
  GnnModelBase.senders
  GnnModelBase.num_edges
  GnnModelBase.num_nodes
)
if (( "${gematria_export_with_annotations}" )); then
  INPUT_TENSORS+=(GraphBuilderModelBase.instruction_node_mask)
fi
if (( "${gematria_export_with_deltas}"
      && "${gematria_export_with_annotations}" )); then
  INPUT_TENSORS+=(TokenGraphBuilderModel.instruction_annotations)
fi
...
readonly INPUT_TENSORS_LIST
```

Same for the output list.